core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/SampleSheet.java

```diff
@@ -0,0 +1,69 @@
+package uk.ac.bbsrc.tgac.miso.core.util;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.eaglegenomics.simlims.core.User;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Index;
+import uk.ac.bbsrc.tgac.miso.core.data.Partition;
+import uk.ac.bbsrc.tgac.miso.core.data.Run;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.PoolableElementView;
+
+public enum SampleSheet {
+  CASAVA_1_7 {
+    @Override
+    protected Stream<String> getColumns() {
+      return Stream.of("FCID", "Lane", "SampleID", "SampleRef", "Index", "Description", "Control", "Recipe", "Operator");
+    }
+
+    @Override
+    protected void makeColumns(Partition p, PoolableElementView dilution, String userName, String[] output) {
+      output[0] = p.getSequencerPartitionContainer().getIdentificationBarcode();
+      output[1] = p.getPartitionNumber().toString();
+      output[2] = String.format("%d_%s_%s", p.getSequencerPartitionContainer().getId(), dilution.getLibraryName(),
+          dilution.getDilutionName());
+      output[3] = dilution.getSampleAlias().replaceAll("\\s", "");
+      output[4] = dilution.getIndices().stream().map(Index::getSequence).collect(Collectors.joining("-"));
+      output[5] = dilution.getLibraryDescription();
+      output[6] = "N";
+      output[7] = "NA";
+      output[8] = userName;
+    }
+  },
+  CASAVA_1_8 {
+    @Override
+    protected Stream<String> getColumns() {
+      return Stream.concat(SampleSheet.CASAVA_1_7.getColumns(), Stream.of("Project"));
+    }
+
+    @Override
+    protected void makeColumns(Partition partition, PoolableElementView dilution, String userName, String[] output) {
+      SampleSheet.CASAVA_1_7.makeColumns(partition, dilution, userName, output);
+      output[9] = dilution.getProjectAlias().replaceAll("\\s", "");
+    }
+  };
+
+  public String createSampleSheet(Run run, User user) {
+    List<String> columns = getColumns().collect(Collectors.toList());
+    return Stream.concat(//
+        Stream.of(String.join(",", columns)), //
+        run.getSequencerPartitionContainers().stream()//
+            .flatMap(container -> container.getPartitions().stream())//
+            .filter(partition -> partition.getPool() != null)//
+            .flatMap(partition -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)